internal/app/poweroff/main.go

```diff
@@ -0,0 +1,39 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
we probably need a way to test this in QEMU if possible

Reply:
I tested it in QEMU by modifying the shutdown call like this:
https://github.com/siderolabs/talos/commit/c5384c46a9c04d594a54686712762edde09d361a

(obviously this is not in the PR)